measure/src/main/scala/org/apache/griffin/measure/job/DQJob.scala

```diff
@@ -25,8 +27,16 @@ case class DQJob(dqSteps: Seq[DQStep]) extends Serializable {
   /**
    * @return execution success
    */
-  def execute(context: DQContext): Boolean = {
-    dqSteps.forall(dqStep => dqStep.execute(context))
+  def execute(context: DQContext): Try[Boolean] = {
+    val tmp = dqSteps.map(dqStep => dqStep.execute(context))
```

Comment:
mapping the elements of `Seq[_]` is not lazy. If there are issues while `.execute`, they will not be handled.

Also, a different variable name can be used in place of `tmp`.

Reply:
I changed the `.execute` method inside and they return a `Try[Boolean]` instance, which wraps the failure potentially be raised.
In the test case added, the spark sql `AnalysisException` is actually raised here from `.execute` method.